hrms/overrides/expense_claim.py

```diff
@@ -0,0 +1,588 @@
+                frappe.throw(_("Employee and its Approver cannot be the same"))
+        else:
+            frappe.throw(_(f"User Id not Found for the  Employee {self.employee}"))
+
+        if self.total_claimed_amount >= RoleConstants.EXPENSE_AMOUNT and not user_has_role(
+            self.approver_1, RoleConstants.EXPENSE_APPROVER2_ROLE
+        ):
+            frappe.throw(
+                _(
+                    f"Select L2 Level Approval since the amount is greater than or equal to {RoleConstants.EXPENSE_AMOUNT}"
+                )
+            )
+        elif not (
+            user_has_role(self.approver_1, RoleConstants.EXPENSE_APPROVER1_ROLE)
+            or user_has_role(self.approver_1, RoleConstants.EXPENSE_APPROVER2_ROLE)
+        ):
+            frappe.throw(
+                _(
+                    f"Approval selected does not have L1 or L2 Expense Approver Role since the amount is less than or equal to {RoleConstants.EXPENSE_AMOUNT}"
+                )
+            )
+
+    def calculate_total_amount(self):
+        self.total_claimed_amount = 0
+        self.total_sanctioned_amount = 0
+
+        for d in self.get("expenses"):
+            self.round_floats_in(d)
+
+            self.total_claimed_amount += flt(d.amount)
+            self.total_sanctioned_amount += flt(d.sanctioned_amount)
+
+        self.round_floats_in(self, ["total_claimed_amount", "total_sanctioned_amount"])
+
+    def validate_sanctioned_amount(self):
+        for d in self.get("expenses"):
+            if flt(d.sanctioned_amount) > flt(d.amount):
+                frappe.throw(
+                    _("Sanctioned Amount cannot be greater than Claim Amount in Row {0}.").format(
+                        d.idx
+                    )
+                )
+
+    def before_insert(self):
+        employee_id = frappe.get_list(
+            "Employee",
+            filters={"user_id": frappe.session.user},
+            pluck="name",
+            ignore_permissions=True,
+        )
+        if frappe.session.user != "Administrator" and (
+            employee_id and employee_id[0] != self.employee
+        ):
+            frappe.throw(_("Cannot Create Expense Claim For Other User"))
+
+    def on_cancel(self):
+        self.status = ExpenseClaimConstants.CANCELLED
+        if self.expense_category_flow == "Flow1":
+            self.state_transtition_check_for_flow1()
+        elif self.expense_category_flow == "Flow2":
+            self.state_transtition_check_for_flow2()
+        self.ignore_linked_doctypes = ("GL Entry", "Stock Ledger Entry", "Payment Ledger Entry")
+        if self.payable_account:
+            self.make_gl_entries(cancel=True)
+        self.update_claimed_amount_in_employee_advance()
+
+    def set_status(self, update=False):
+        pass
+
+    def on_update(self):
+        pass
+
+    def on_submit(self):
+        if self.is_submit_and_cancel:
+            self.cancel()
+        else:
+            self.make_gl_entries()
```

Comment:
move to state transition on HR L2 Approval

Reply:
Done.